### ISSUE TITLE
ls on on the path where the blobfuse is mounted hangs

### ISSUE THREAD

**Issue by 123dev** [Azure/azure-storage-fuse#600]
### Which version of the blobfuse was used?
blobfuse 1.3.7

### Which OS (please include version) are you using?
Ubuntu 20.04
in docker under wsl2
Linux 9f626158259e 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### What problem was encountered?
When I issue the blobfuse command, I get no errors and the volume is seemingly mounted.
The command used to mount:
```
blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_
other
```

After seemingly successful mount, if I do `ls -l /root`
It hangs  a very long time and it eventually lists the directory, however I suspect at that point the volume is unmounted.
I say that because:
1- `ls -l root` immediately returns and `ls -al /root/source` is empty

2- While the ls is hung, If I try to list the contents of /mnt/blobfusetmp (from another terminal)
I see
root/
However once the other command times out and returns, the contents of blobfusetmp is empty (no more root/)

So I suspect the volume is getting mounted but some other problems are encountered, and then eventually the volume is unmounted.

### Have you found a mitigation/solution?
No

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
no Syslog, using casandra docker image

### If relevant, please share your mount command.
```
blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_
other
```

Thanks

**Comment by vibhansa-msft** [comment]
If observation is /mnt/blobfusetmp/root is removed after some time then indeed its getting unmounted. We have observed 'ls -l' command getting hung up when there are connection timeouts and blobfuse is not able to connect to Azure. To root-cause this we need syslogs in LOG_DEBUG mode. In case your syslog isn't working, is it possible to install 'rsyslog' service explicitly in your wsl.

**Comment by 123dev** [comment]
OK thanks @vibhansa-msft 
I'll try to install it and do my best to capture the logs.

**Comment by 123dev** [comment]
@vibhansa-msft 
As mentioned previously, the mount is attempted inside a docker container
https://hub.docker.com/_/cassandra

I'm not sure what I need to install where and configure what to capture the logs that you require.
Some guidance or pointers would be appreciated.

```
uname -a
Linux 9f626158259e 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
Linux Debian bullseye/sid ( 5.4.72-microsoft-standard-WSL2 x86_64)
```

**Comment by vibhansa-msft** [comment]
You need to install rsyslog service using 'sudo apt-get install rsyslog -y'. Once this installs correctly there are two config files you need to copy, '10-blobfuse.conf' and 'blobfuse-logrotate'. These files can be downloaded from 'azure-storage-fuse/systemd/' directory in the repo. Use below command to copy them to their respective locations:
     sudo cp ./systemd/10-blobfuse.conf /etc/rsyslog.d/
     sudo cp ./systemd/blobfuse-logrotate /etc/logrotate.d/blobfuse
Once copy is done do 'sudo service rsyslog restart'.
Now mount your blobfuse and you can find logs in /var/log/blobfuse.log.

**Comment by 123dev** [comment]
Thanks @vibhansa-msft 

Here's the log file.
```
May 17 11:20:55 9f626158259e blobfuse[1756]: Auth type set as sas#015
May 17 11:20:55 9f626158259e blobfuse[1756]: Auth Type set to SAS_AUTH
May 17 11:20:55 9f626158259e blobfuse[1756]: Account name found
May 17 11:20:55 9f626158259e blobfuse[1756]: Sas Token found
May 17 11:20:55 9f626158259e blobfuse[1756]: Container name found
May 17 11:20:55 9f626158259e blobfuse[1756]: Setting logging level to : LOG_WARNING
May 17 11:20:55 9f626158259e blobfuse[1756]: ** Delaying tls init to post fork for older libcurl version
May 17 11:20:55 9f626158259e blobfuse[1756]: ** Delaying authentication to post fork for older curl versions
May 17 11:20:55 9f626158259e blobfuse[1757]: ** Post fork authentication for older libcurl version
May 17 11:20:55 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:20:55 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:20:56 9f626158259e blobfuse[1757]: message repeated 5 times: [  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:20:55 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 :: ]
May 17 11:20:57 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:20:56 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:20:58 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:20:57 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:00 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:20:58 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:02 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:00 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:06 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:02 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:11 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:06 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:17 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:11 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:24 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:17 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:33 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:24 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:21:47 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:33 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:22:04 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:21:47 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:22:22 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:22:04 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:22:45 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:22:22 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:23:06 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:22:45 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:23:38 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:23:06 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:24:06 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:23:38 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:24:45 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:24:06 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:25:29 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:24:45 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:26:38 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:25:29 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:27:27 9f626158259e blobfuse[1757]:  ==> REQUEST/RESPONSE :: GET https://REDACTED#015.blob.core.windows.net/REDACTED%0D?comp=list&delimiter=/&include=metadata&maxresults=1&restype=container&sv=2020-04-08&ss=btqf&srt=sco&st=2021-05-12T12%3A25%3A47Z&se=2022-05-13T12%3A25%3A00Z&sp=rfl&sig=REDACTED?&User-Agent=azure-storage-fuse/1.3.7&x-ms-date=Mon, 17 May 2021 11:26:38 GMT&x-ms-version=2018-11-09&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 0 ::
May 17 11:27:27 9f626158259e blobfuse[1757]: Unable to start blobfuse. Failed to connect to the storage container. There might be something wrong about the storage config, please double check the storage account name, account key/sas token/OAuth access token and container name. errno=3
May 17 11:27:27 9f626158259e blobfuse[1757]: Unable to start blobfuse due to a lack of credentials. Please check the readme for valid auth setups.
May 17 11:27:27 9f626158259e blobfuse[1757]: Unmounting blobfuse.
May 17 11:27:27 9f626158259e blobfuse[1757]: Unmounted blobfuse successfully.
```

**Comment by 123dev** [comment]
@vibhansa-msft 

You should be able to reproduce the issue with the following Dockerfile
```
FROM cassandra

RUN apt-get -qqy update && apt-get -qqy install --no-install-recommends \
    vim \
    wget

# Add the Microsoft packages repo
RUN wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb
# Download blobfuse syslog files for debugging,
RUN wget https://raw.githubusercontent.com/Azure/azure-storage-fuse/master/systemd/10-blobfuse.conf
RUN wget https://raw.githubusercontent.com/Azure/azure-storage-fuse/master/systemd/blobfuse-logrotate
# Install the microsoft packages repo
RUN dpkg -i packages-microsoft-prod.deb
# Install blobfuse and other dependecies
RUN apt-get -qqy update && apt-get -qqy install --no-install-recommends \
    blobfuse \
    fuse \
    rsyslog \
    libcurl3-gnutls

# Create the blobfuse temporary directory
RUN mkdir /mnt/blobfusetmp -p

# Copy the credentials / configuration file
COPY ["./fuse_connection.cfg", "/root/.fuse_connection.cfg"]
RUN chmod 600 /root/.fuse_connection.cfg
# Create the mount directory
RUN mkdir /root/source

# Copy syslog debugging files to the proper location
RUN cp ./10-blobfuse.conf /etc/rsyslog.d/10-blobfuse.conf
RUN cp ./blobfuse-logrotate /etc/rsyslog.d/blobfuse-logrotate
# Restart the rsyslog service to load the above configuration
RUN service rsyslog restart

# The command to run to mount the blob
# RUN blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other
```

Create a file `fuse_connection.cfg` in the same Directory as the Dockerfile with the following (update to use your blob)
```
authType SAS
accountName exmapleblob
sasToken sv=SAS_TOKEN
containerName exampleContainer
```

PS: I'm able to mount the same blob following same process in another container based on `mcr.microsoft.com/powershell:latest` image.
But we need it mounted in a container based on `casandra` image which is having the above issue.

Let me know if you need anything further.

**Comment by NaraVen** [comment]
your url encoding appears incompatible with Curl. From scratching the surface it looks like a windows LF issue in the config file. Could you please do a  dos2unix of your fuse_connection.cfg file and try again

**Comment by 123dev** [comment]
@NaraVen 

Thank you, indeed it was the case, I'm now able to mount properly and access it.
Many thanks.

It would be a nice to have if blobfuse handled the line endings properly, or at least warned about it.

Many thanks

**Comment by NaraVen** [comment]
@123dev, Blobfuse cannot say if it is indeed a character or a line ending. It will take what Linux gives. Glad it works.

**Comment by 123dev** [comment]
Now that I could mount the blob while in the container,
Un-commenting this line in the Dockerfile is causing a segmentation fault.

```
RUN blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other
```




```
 => ERROR [15/15] RUN blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o  0.5s
------
 > [15/15] RUN blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other:
#19 0.453 Segmentation fault
------
executor failed running [/bin/sh -c blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other]: exit code: 139
```

Any ideas as to why?

Thanks?

**Comment by 123dev** [comment]
Adding 30 seconds delay did not help

What would be the fstab equivalent of this if I want to mount it through fstab?
```
/usr/bin/blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other
```

Thanks

**Comment by NaraVen** [comment]
what do the logs say ? docker inspect [containerID] will tell you what went wrong.  It could be a memory issue or a lack of privilege issue. yes, you can /etc/fstab and add this line blobfuse /home/nara/mntblobfuse fuse delay_connect_netdev,--tmp-path=/home/nara/tmppath,--config-file=/home/nara/connection.cfg,attr_timeout=240,entry_timeout=240,negative_timeout=120,allow_other 0 0

**Comment by 123dev** [comment]
Thanks @NaraVen 
This is happening at build time, container is not even created.
```
Creating network "casandra_default" with the default driver
Building cassandra-seed
[+] Building 2.2s (21/21) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                   0.0s
 => => transferring dockerfile: 1.88kB                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/cassandra:latest                                                                                                                    1.8s
 => [ 1/17] FROM docker.io/library/cassandra@sha256:7d7890a57cad7b104ff9c77eb140af95730bd6c360fe24681d92834b1ae73bdf                                                                   0.0s
 => [internal] load build context                                                                                                                                                      0.0s
 => => transferring context: 70B                                                                                                                                                       0.0s
 => CACHED [ 2/17] RUN apt-get -qqy update && apt-get -qqy install --no-install-recommends     vim     wget                                                                            0.0s
 => CACHED [ 3/17] RUN wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb                                                                             0.0s
 => CACHED [ 4/17] RUN wget https://raw.githubusercontent.com/Azure/azure-storage-fuse/master/systemd/10-blobfuse.conf                                                                 0.0s
 => CACHED [ 5/17] RUN wget https://raw.githubusercontent.com/Azure/azure-storage-fuse/master/systemd/blobfuse-logrotate                                                               0.0s
 => CACHED [ 6/17] RUN dpkg -i packages-microsoft-prod.deb                                                                                                                             0.0s
 => CACHED [ 7/17] RUN apt-get -qqy update && apt-get -qqy install --no-install-recommends     blobfuse     fuse     rsyslog     libcurl3-gnutls                                       0.0s
 => CACHED [ 8/17] RUN mkdir /mnt/blobfusetmp -p                                                                                                                                       0.0s
 => CACHED [ 9/17] COPY [./fuse_connection3.cfg, /root/.fuse_connection.cfg]                                                                                                           0.0s
 => CACHED [10/17] RUN chmod 600 /root/.fuse_connection.cfg                                                                                                                            0.0s
 => CACHED [11/17] COPY [./mount.sh, /root/mount.sh]                                                                                                                                   0.0s
 => CACHED [12/17] RUN chmod 755 /root/mount.sh                                                                                                                                        0.0s
 => CACHED [13/17] RUN mkdir /root/source                                                                                                                                              0.0s
 => CACHED [14/17] RUN cp ./10-blobfuse.conf /etc/rsyslog.d/10-blobfuse.conf                                                                                                           0.0s
 => CACHED [15/17] RUN cp ./blobfuse-logrotate /etc/rsyslog.d/blobfuse-logrotate                                                                                                       0.0s
 => CACHED [16/17] RUN service rsyslog restart                                                                                                                                         0.0s
 => ERROR [17/17] RUN /usr/bin/blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeo  0.3s
------
 > [17/17] RUN /usr/bin/blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other:
#21 0.234 Segmentation fault
------
executor failed running [/bin/sh -c /usr/bin/blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other]: exit code: 139
ERROR: Service 'cassandra-seed' failed to build : Build failed
```

If I comment out the line 
```
RUN /usr/bin/blobfuse /root/source --tmp-path=/mnt/blobfusetmp  --config-file=/root/.fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other
```
Then  it builds and runs and when I do docker exec to it and run that same command, it gets mounted.

Weird.

**Comment by 123dev** [comment]
if I replace the manual mount line with this
```
RUN echo "/usr/bin/blobfuse /root/source fuse _netdev,--tmp-path=/mnt/blobfusetmp,--config-file=/root/.fuse_connection.cfg,attr_timeout=240,entry_timeout=240,negative_timeout=120,allow_other 0 0" >> /etc/fstab
```

The container runs, but the volume is not mounted automatically,
if I docker exec to the container and do mount -a, it gets mounted,
somehow fuse did not like `delay_connect`
```
fuse: unknown option `delay_connect'
```

and if I include the mount -a in the build process I still get the seg fault
```
 => CACHED [16/18] RUN service rsyslog restart                                                                                                                                         0.0s
 => CACHED [17/18] RUN echo "/usr/bin/blobfuse /root/source fuse _netdev,--tmp-path=/mnt/blobfusetmp,--config-file=/root/.fuse_connection.cfg,attr_timeout=240,entry_timeout=240,nega  0.0s
 => ERROR [18/18] RUN mount -a                                                                                                                                                         0.3s
------
 > [18/18] RUN mount -a:
#22 0.243 Segmentation fault
------
executor failed running [/bin/sh -c mount -a]: exit code: 32
ERROR: Service 'cassandra-seed' failed to build : Build failed
```

**Comment by NaraVen** [comment]
remove delay_connect, It doesn't work on Ubuntu, you are right.  Regarding running blobfuse in the build it is a privileged operation i am not sure it can be run in the build.

**Comment by 123dev** [comment]
Thanks @NaraVen 
Why blobfuse in build is not possible?

Running blobfuse with this particular container is not going to be simple, I cannot just simply alter the CMD, it uses the CMD as a parameter to the [ENTRYPOINT](https://github.com/docker-library/cassandra/blob/master/docker-entrypoint.sh), 
The default CMD is:
[CMD ["cassandra", "-f"]](https://github.com/docker-library/cassandra/blob/f00a725cc0189ef166ac1d893227651bd81a6996/Dockerfile.template#L171)

Any other suggestions?
Thanks

**Comment by NaraVen** [comment]
you can try adding CMD["blobfuse",....] in the .dockerfile and see if it works. Otherwise docker exec is your only option. A container's lifecycle is 1 application, it ends with the application. So RUN blobfuse will have issues accessing the memory. That's why the segmentation fault.
